core/embed/rust/src/ui/model_tt/component/simple_page.rs

```diff
@@ -64,23 +83,42 @@ where
         // paint.
         self.fade = Some(theme::BACKLIGHT_NORMAL);
     }
+
+    fn is_horizontal(&self) -> bool {
+        matches!(self.axis, Axis::Horizontal)
+    }
 }
 
-impl<T> Component for HorizontalPage<T>
+impl<T> Component for SimplePage<T>
 where
     T: Paginate,
     T: Component,
 {
-    type Msg = PageMsg<T::Msg, Never>;
+    type Msg = PageMsg<T::Msg>;
 
     fn place(&mut self, bounds: Rect) -> Rect {
         self.swipe.place(bounds);
 
-        let (content, scrollbar) = bounds.split_bottom(SCROLLBAR_HEIGHT + SCROLLBAR_BORDER);
-        self.pad.place(content);
-        self.content.place(content);
-        self.scrollbar
-            .place(scrollbar.inset(Insets::bottom(SCROLLBAR_BORDER)));
+        let (content, scrollbar) = if self.is_horizontal() {
+            bounds.split_bottom(SCROLLBAR_HEIGHT + SCROLLBAR_BORDER)
+        } else {
+            bounds.split_right(SCROLLBAR_HEIGHT + SCROLLBAR_BORDER)
+        };
+
+        self.content.place(bounds);
+        self.pad.place(bounds);
```

Comment:
Moved the pad placement in [8971906](https://github.com/trezor/trezor-firmware/pull/3307/commits/8971906b431e16e53a07d9a2cf2264353ab857d4), however content must be placed two times because the operation changes the number of pages.